pyroomacoustics/libroom_src/room.cpp

```diff
@@ -0,0 +1,1026 @@
+    float _sound_speed,
+    // parameters for the image source model
+    int _ism_order,
+    // parameters for the ray tracing
+    float _energy_thres,
+    float _time_thres,
+    float _mic_radius,
+    float _mic_hist_res,
+    bool _is_hybrid_sim
+    )
+  : microphones(_microphones),
+  sound_speed(_sound_speed), ism_order(_ism_order), energy_thres(_energy_thres), time_thres(_time_thres),
+  mic_radius(_mic_radius), mic_hist_res(_mic_hist_res), is_hybrid_sim(_is_hybrid_sim),
+  is_shoebox(true), shoebox_size(_room_size), shoebox_absorption(_absorption)
+{
+  if (shoebox_absorption.rows() != _scattering.rows())
+  {
+    throw std::runtime_error("Error: The same number of absorption and scattering coefficients are reqquired");
+  }
+
+  make_shoebox_walls(shoebox_size, _absorption, _scattering);
+  init();
+}
+
+
+template<>
+void Room<2>::make_shoebox_walls(
+    const Vectorf<2> &rs,  // room_size
+    const Eigen::Array<float,Eigen::Dynamic,4> &abs,
+    const Eigen::Array<float,Eigen::Dynamic,4> &scat
+    )
+{
+  Eigen::Matrix<float,2,Eigen::Dynamic> corners;
+  corners.resize(2, 2);
+
+  corners << 0.f, rs[0], 0.f, 0.f;
+  walls.push_back(Wall<2>(corners, abs.col(2), scat.col(2), "south"));
+
+  corners << rs[0], rs[0], 0.f, rs[1];
+  walls.push_back(Wall<2>(corners, abs.col(1), scat.col(1), "east"));
+
+  corners << rs[0], 0.f, rs[1], rs[1];
+  walls.push_back(Wall<2>(corners, abs.col(3), scat.col(3), "north"));
+
+  corners << 0.f, 0.f, rs[1], 0.f;
+  walls.push_back(Wall<2>(corners, abs.col(0), scat.col(0), "west"));
+}
+
+
+template<>
+void Room<3>::make_shoebox_walls(
+    const Vectorf<3> &rs,  // room_size
+    const Eigen::Array<float,Eigen::Dynamic,6> &abs,
+    const Eigen::Array<float,Eigen::Dynamic,6> &scat
+    )
+{
+  Eigen::Matrix<float,3,Eigen::Dynamic> corners;
+  corners.resize(3,4);
+
+  corners << 0.f, 0.f, 0.f, 0.f,
+             rs[1], 0.f, 0.f, rs[1],
+             0.f, 0.f, rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(0), scat.col(0), "west"));
+
+  corners << rs[0], rs[0], rs[0], rs[0], 
+             0.f, rs[1], rs[1], 0.f, 
+             0.f, 0.f, rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(1), scat.col(1), "east"));
+
+  corners << 0.f, rs[0], rs[0], 0.f,
+             0.f, 0.f, 0.f, 0.f, 
+             0.f, 0.f, rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(2), scat.col(2), "south"));
+
+  corners << rs[0], 0.f, 0.f, rs[0],
+             rs[1], rs[1], rs[1], rs[1],
+             0.f, 0.f, rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(3), scat.col(3), "north"));
+
+  corners << rs[0], 0.f, 0.f, rs[0],
+             0.f, 0.f, rs[1], rs[1],
+             0.f, 0.f, 0.f, 0.f;
+  walls.push_back(Wall<3>(corners, abs.col(4), scat.col(4), "floor"));
+
+  corners << rs[0], rs[0], 0.f, 0.f,
+             0.f, rs[1], rs[1], 0.f,
+             rs[2], rs[2], rs[2], rs[2];
+  walls.push_back(Wall<3>(corners, abs.col(4), scat.col(4), "ceiling"));
```

Comment:
should this be using `abs.col(5)` and `scat.col(5)`? or is it assumed floor and ceiling have same coefficients?

Reply:
Oh! Nice save, thanks! Obviously, it should be `abs.col(5)` and `scat.col(5)`!